atoma-daemon/src/lib.rs

```diff
@@ -1,3 +1,11 @@
+#![allow(clippy::cast_possible_truncation)]
+#![allow(clippy::cast_possible_wrap)]
+#![allow(clippy::cast_precision_loss)]
+#![allow(clippy::missing_docs_in_private_items)]
+#![allow(clippy::doc_markdown)]
+#![allow(clippy::module_name_repetitions)]
+#![allow(clippy::cast_sign_loss)]
```

Comment:
These will require quite significant changes, the most substantive being related to 

```rust 
#![allow(clippy::cast_sign_loss)]
#![allow(clippy::cast_possible_truncation)]
#![allow(clippy::cast_possible_wrap)]
#![allow(clippy::cast_precision_loss)]
```

I've outlined the issue more in https://github.com/atoma-network/atoma-node/issues/374